### PR TITLE
Cut 0.0.105

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Still waiting on a few more things, but getting a bit ahead of this one now.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Updated to latest upstream, which should be basically complete.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,102 @@
+# 0.0.105 - 2022-02-28
+
+## API Updates
+ * `Phantom node` payments are now supported, allowing receipt of a payment on
+   any one of multiple nodes without any coordination across the nodes being
+   required. See the new `PhantomKeysManager`'s docs for more, as well as
+   requirements on `KeysInterface::get_inbound_payment_key_material` and
+   `lightning_invoice::utils::create_phantom_invoice` (#1199).
+ * In order to support phantom node payments, several `KeysInterface` methods
+   now accept a `Recipient` parameter to select between the local node_id and a
```
Comment (jkczyz) [inline CHANGELOG.md:10]: tick `node_id`

**Review by valentinewallace — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,102 @@
...
+   the `BOLT 3` secret storage scheme (#1299).
+ * Several `Sign` methods now receive HTLC preimages as proof of state
+   transition, see new documentation for more (#1251).
+ * `KeysInterface::sign_invoice` now provides the HRP and other invoice data
+   separately to make it simpler for external signers to parse (#1272).
+ * `Sign::sign_channel_announcement` now returns both the node's signature and
+   the per-channel signature. `InMemorySigner` now requires the node's secret
+   key in order to implement this (#1179).
+ * `ChannelManager` deserialization will now fail if the `KeysInterface` used
+   has a different node_id than the `ChannelManager` expects (#1250).
```
Comment (jkczyz) [inline CHANGELOG.md:40]: tick `node_id`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,102 @@
...
+ * Fixed cases where slow code (including user I/O) could cause us to
+   disconnect peers with ping timeouts in `BackgroundProcessor` (#1269).
+ * Now persist the `ChannelManager` prior to `BackgroundProcessor` stopping,
+   preventing race conditions where channels are closed on startup even with a
+   clean shutdown. This requires that users stop network processing and
+   disconnect peers prior to `BackgroundProcessor` shutdown (#1253).
+ * Fields in `ChannelHandshakeLimits` provided via the `override_config` to
+   `create_channel` are now applied instead of the default config (#1292).
+ * Fixed the generation of documentation on docs.rs to include API surface
+   which is hidden behind feature flags (#1303).
```
Comment (jkczyz) [inline CHANGELOG.md:57]: nit: s/surface which is/surfaces that are

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,102 @@
...
+ * Moved more instances of verbose gossip logging from the `Trace` level to the
+   `Gossip` level (#1220).
+ * Delayed `announcement_signatures` until the channel has six confirmations,
+   slightly improving propagation of channel announcements (#1179).
+ * Several fixes in script and transaction weight calculations when anchor
+   outputs are enabled (#1229).
+
+## Serialization Compatibility
+ * Using `ChannelManager` data written by versions prior to 0.0.105 will result
+   in preimages for HTLCs which were pending at startup to be missing in calls
```
Comment (jkczyz) [inline CHANGELOG.md:74]: s/which/that

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,102 @@
...
+   to `KeysInterface` methods (#1251).
+ * Any phantom invoice payments received on a node which is not upgraded to
```
Comment (jkczyz) [inline CHANGELOG.md:76]: s/which/that

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,102 @@
...
+ * Elias Rohrer
+ * Jeffrey Czyz
+ * Jurvis Tan
+ * Ken Sedgwick
+ * Matt Corallo
+ * Naveen
+ * Tibo-lg
+ * Valentine Wallace
+ * Viktor Tigerström
+ * dependabot[bot]
```
Comment (jkczyz) [inline CHANGELOG.md:94]: lol

**Review by jkczyz — APPROVED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (TheBlueMatt) [reply CHANGELOG.md:94]: Yea....I guess we should leave it in? I mean....it *did* contribute, right?

**Review by jkczyz — DISMISSED** [review]

**Comment by TheBlueMatt** [comment]
Squashed without changes.

**Review by valentinewallace — APPROVED** [review]

**Review by jkczyz — APPROVED** [review]

**Pull request merged**